### PR TITLE
Test new high granularity Tracker Alignment PCL

### PR DESCRIPTION
**Pull request by tvami**
# Replay Request

**Requestor**  

AlCaDB

**Describe the configuration**  
* Release: CMSSW_12_4_1
* Run: ~352929~, then 324841
* GTs:
   * expressGlobalTag: ~124X_dataRun3_Express_v3~, then 124X_dataRun3_Express_TIER0_REPLAY_Run2_v1
   * promptrecoGlobalTag: ~124X_dataRun3_Prompt_v3~, then 124X_dataRun3_Prompt_TIER0_REPLAY_Run2_v1
   * alcap0GlobalTag: ~124X_dataRun3_Prompt_v3~, then 124X_dataRun3_Prompt_TIER0_REPLAY_Run2_v1
* Additional changes: Add new high granularity (HG) Tracker Alignment PCL (`PromptCalibProdSiPixelAliHG`) to the rest of express

**Purpose of the test**  

Test HG Tracker Alignment PCL

**T0 Operations cmsTalk thread**  
If necessary, provide a link to the cmsTalk thread announcing the test to the relevant groups. 
[Tier0 Operations cmsTalk Forum](https://cms-talk.web.cern.ch/c/offcomp/compops/tier0/210?)

### CODE REVIEW

**Comment by mmusich** [comment]
@dmeuser @mteroerd @connorpa @antoniovagnerini @consuegs FYI

**Comment by tvami** [comment]
test syntax please

**Comment by jhonatanamado** [comment]
Hi guys, here the monitoring link for this replay  https://monit-grafana.cern.ch/d/t_jr45h7k/cms-tier0-replayid-monitoring?orgId=11&var-Bin=5m&var-ReplayID=220701154140&var-JobType=All&var-WorkflowType=All

**Comment by jhonatanamado** [comment]
replay finished without issues. Output data is the following:

```
/StreamExpress/Tier0_REPLAY_2022-Express-v464/DQMIO#ffa01382-c672-4a56-9c19-8543e52c8eab,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiStripGains-Express-v464/ALCAPROMPT#2ee27fc4-6345-4a91-82c4-4ef2f1795982,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiStripHitEff-Express-v464/ALCAPROMPT#8a3df882-2546-4b0a-a58d-013346713c94,
/StreamExpress/Tier0_REPLAY_2022-SiStripPCLHistos-Express-v464/ALCARECO#50fee08b-f7b9-4432-ba92-1f9d6dd685e2,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiPixel-Express-v464/ALCAPROMPT#820d84e7-8398-4abd-8723-530f933bf6f7,
/StreamExpress/Tier0_REPLAY_2022-SiStripCalMinBiasAAG-Express-v464/ALCARECO#32c84280-a921-47cd-b173-83a67d878aa3,
/StreamExpress/Tier0_REPLAY_2022-SiStripCalZeroBias-Express-v464/ALCARECO#7c5d23d7-f7b6-4f80-97f7-262334355e74,
/ExpressPhysics/Tier0_REPLAY_2022-Express-v464/FEVT#be2f3a16-28e2-4da1-9239-89cbfd07bcf2,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiPixelLA-Express-v464/ALCAPROMPT#101e3417-f25e-434c-943b-8f774fea13dd,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProd-Express-v464/ALCAPROMPT#76089597-781a-4018-ae86-bc9c9a7b3a2f,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiPixelAli-Express-v464/ALCAPROMPT#ec51dcea-ca69-4433-a165-341f4c2506e4,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiPixelAliHG-Express-v464/ALCAPROMPT#07d38b6b-3590-4c89-b302-c64c7383a879,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiStrip-Express-v464/ALCAPROMPT#f199871e-59c6-46f0-8a27-11ade5fc9bee,
/StreamExpress/Tier0_REPLAY_2022-SiStripCalMinBias-Express-v464/ALCARECO#a69c0bbf-8607-48fa-a356-6c9ec8d1736d,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiStripGainsAAG-Express-v464/ALCAPROMPT#2befe9be-421f-4f5d-be07-8d36ff12e1af,
/StreamExpress/Tier0_REPLAY_2022-SiPixelCalSingleMuon-Express-v464/ALCARECO#c3ff2fd1-cc6e-4e13-af5a-edaba51ea6c5,
/StreamExpress/Tier0_REPLAY_2022-SiPixelCalZeroBias-Express-v464/ALCARECO#f0a11404-4bc3-42b0-8a10-c66c975ca547,
/StreamExpress/Tier0_REPLAY_2022-TkAlMinBias-Express-v464/ALCARECO#b482b36e-a23b-4191-9f8d-581b78fbe4f9,
/StreamExpress/Tier0_REPLAY_2022-Express-v464/DQMIO#91bcd8d1-d7d6-4f13-bc3a-3575528c14e6,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiStripGains-Express-v464/ALCAPROMPT#cf33794e-9821-40ef-83c7-521f165bf0c2,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiStripHitEff-Express-v464/ALCAPROMPT#7c9aa4ce-6edd-44fc-8dc8-2b55b6e62346,
/StreamExpress/Tier0_REPLAY_2022-SiStripPCLHistos-Express-v464/ALCARECO#7b95c2b2-bdbe-4f2d-8022-f03508f22834,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiPixel-Express-v464/ALCAPROMPT#47db29af-88e0-4ff9-a9bf-f2aea6136beb,
/StreamExpress/Tier0_REPLAY_2022-SiStripCalMinBiasAAG-Express-v464/ALCARECO#783f3b5c-410c-4e37-bb86-70efe3ebead2,
/StreamExpress/Tier0_REPLAY_2022-SiStripCalZeroBias-Express-v464/ALCARECO#046ff52a-ebe0-431e-863c-e1002ec9a2fd,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiPixelLA-Express-v464/ALCAPROMPT#6207d468-8073-4fb8-8401-483e29487ee0,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProd-Express-v464/ALCAPROMPT#bfe3af35-e937-4417-8523-79da77bf4fb3,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiPixelAli-Express-v464/ALCAPROMPT#2b31aae1-3f5c-4422-a41e-e63ed6168728,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiPixelAliHG-Express-v464/ALCAPROMPT#8c26699e-6bc8-4b04-8892-1b8752305d9a,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiStrip-Express-v464/ALCAPROMPT#336548a8-a31a-46de-bec1-791496e3f525,
/StreamExpress/Tier0_REPLAY_2022-SiStripCalMinBias-Express-v464/ALCARECO#a9a3af4d-4c9b-4ee4-bbd9-076fabaa7b00,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiStripGainsAAG-Express-v464/ALCAPROMPT#8b9e4ea7-fbf1-419b-8683-afcb4116ddbe,
/StreamExpress/Tier0_REPLAY_2022-SiPixelCalSingleMuon-Express-v464/ALCARECO#3dec41f2-07fb-41aa-af00-d7ecb6b3e21f,
/StreamExpress/Tier0_REPLAY_2022-SiPixelCalZeroBias-Express-v464/ALCARECO#c5cb42e6-0eb3-4b6d-a760-b4e0fffe7857,
/StreamExpress/Tier0_REPLAY_2022-TkAlMinBias-Express-v464/ALCARECO#6b13b7b2-7d2d-4ff7-9696-060364198842,
/ExpressPhysics/Tier0_REPLAY_2022-Express-v464/FEVT#f89667f2-b531-4301-8d9c-886cda485590,
/ExpressPhysics/Tier0_REPLAY_2022-Express-v464/FEVT#46acd44d-dbe2-4ba0-b79e-05502959dbde,
/StreamExpress/Tier0_REPLAY_2022-Express-v464/DQMIO#b6aba70d-e127-403b-bed9-52a7ab99ef68,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiStripGains-Express-v464/ALCAPROMPT#c467fc0f-f507-4809-a292-87e6e1f5d131,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiStripHitEff-Express-v464/ALCAPROMPT#7dcd963a-08e1-4192-a86a-675e0e224a48,
/StreamExpress/Tier0_REPLAY_2022-SiStripPCLHistos-Express-v464/ALCARECO#10d43998-0135-4063-aa26-59d2af2c63cf,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiPixel-Express-v464/ALCAPROMPT#1b937982-82f6-48cc-8089-e4a54f9392a9,
/StreamExpress/Tier0_REPLAY_2022-SiStripCalMinBiasAAG-Express-v464/ALCARECO#aee93b45-32b1-4a65-b071-9bb808ff2c8d,
/StreamExpress/Tier0_REPLAY_2022-SiStripCalZeroBias-Express-v464/ALCARECO#9b6239e2-72c4-4710-84d4-6ffcb46282ed,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiPixelLA-Express-v464/ALCAPROMPT#6ccf7b83-1414-4e2a-8bba-288adb38b932,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProd-Express-v464/ALCAPROMPT#e72c74d6-f64d-4dda-bbb6-6c1f12414be2,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiPixelAli-Express-v464/ALCAPROMPT#306b3257-6f71-43ce-b41d-497c5b978fd4,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiPixelAliHG-Express-v464/ALCAPROMPT#11e13fd5-c84f-4d0e-9e5b-cbf4d82e6c66,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiStrip-Express-v464/ALCAPROMPT#eac1b01a-3b18-4efc-8510-9ee7e03f532d,
/StreamExpress/Tier0_REPLAY_2022-SiStripCalMinBias-Express-v464/ALCARECO#87a646dd-415c-4966-9b01-1a29a585abb4,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiStripGainsAAG-Express-v464/ALCAPROMPT#0d71ba95-e78c-46e0-bb0d-97a8739bf482,
/StreamExpress/Tier0_REPLAY_2022-SiPixelCalSingleMuon-Express-v464/ALCARECO#38fcad0c-6102-47b3-9333-357a50ec9af7,
/StreamExpress/Tier0_REPLAY_2022-SiPixelCalZeroBias-Express-v464/ALCARECO#e5f9ccda-d0de-425c-9235-475eca8a0a45,
/StreamExpress/Tier0_REPLAY_2022-TkAlMinBias-Express-v464/ALCARECO#2e202bca-4664-4853-b5cc-fb20f5ad8635,
/ExpressPhysics/Tier0_REPLAY_2022-Express-v464/FEVT#a464e5fc-632a-4086-8889-383dc5142b8b,
/StreamExpress/Tier0_REPLAY_2022-Express-v464/DQMIO#1bcbe4eb-be3e-4370-80c2-ff30fadcc2d6,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiStripGains-Express-v464/ALCAPROMPT#1ade4c53-a29e-478f-9a82-c3116fc9ae1f,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiStripHitEff-Express-v464/ALCAPROMPT#f8eedfdc-354f-4652-9ca1-c864ffe8ece3,
/StreamExpress/Tier0_REPLAY_2022-SiStripPCLHistos-Express-v464/ALCARECO#a1cc8fa1-1be9-4644-8174-53824d045bb8,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiPixel-Express-v464/ALCAPROMPT#33ff7595-2db9-4e92-a4e2-2d7aabeace31,
/StreamExpress/Tier0_REPLAY_2022-SiStripCalMinBiasAAG-Express-v464/ALCARECO#b5cc293c-54e8-4076-9f43-676c6441bd23,
/StreamExpress/Tier0_REPLAY_2022-SiStripCalZeroBias-Express-v464/ALCARECO#bf1fb0b7-d4b2-42f3-a691-d8eba71f1911,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiPixelLA-Express-v464/ALCAPROMPT#321dbe27-333e-499b-8a4d-9d67f04097c2,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProd-Express-v464/ALCAPROMPT#4f9ee761-bbd2-4622-9f06-00de99b0a367,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiPixelAli-Express-v464/ALCAPROMPT#cfca6270-c89f-4c16-8873-445d9a35f0ee,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiPixelAliHG-Express-v464/ALCAPROMPT#de6f97b3-4ce5-4a3d-88c7-540505fa77d2,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiStrip-Express-v464/ALCAPROMPT#ac14f310-6f56-43a1-bc48-a8699dc17faa,
/StreamExpress/Tier0_REPLAY_2022-SiStripCalMinBias-Express-v464/ALCARECO#3cef3746-59d6-47be-ba28-0f1025518d5f,
/StreamExpress/Tier0_REPLAY_2022-PromptCalibProdSiStripGainsAAG-Express-v464/ALCAPROMPT#0eb7a762-06c6-4b52-99a1-b442795f4bf3,
/StreamExpress/Tier0_REPLAY_2022-SiPixelCalSingleMuon-Express-v464/ALCARECO#7c1681e4-0218-49c8-82e7-b116448b13ae,
/StreamExpress/Tier0_REPLAY_2022-SiPixelCalZeroBias-Express-v464/ALCARECO#17fe8afa-3c07-4d02-8d09-e4a5c840d061,
/StreamExpress/Tier0_REPLAY_2022-TkAlMinBias-Express-v464/ALCARECO#a4908ba1-0a54-4f62-ab77-3200619bbbf7,
```

**Comment by mmusich** [comment]
Hi @jhonatanamado thanks for the update.
While the replay did finish without issues, the output of the HG PCL Alignment does not look good: http://tinyurl.com/2cmbvhu3

Screenshot from 2022-07-02 10-58-57

(the regular LG alignment ended fine http://tinyurl.com/28rhsvhu)

@dmeuser @mteroerd can you please have a look?

@tvami the run chosen for this replay is a bit low on the stats side, can we try a 2018 run if there are still streamers?

**Comment by tvami** [comment]
> the run chosen for this replay is a bit low on the stats side, can we try a 2018 run if there are still streamers?

Hi Marco, at this point, I'd rather wait a few more days and use a 13.6 TeV run for the test. What do you think?

**Comment by mmusich** [comment]
> at this point, I'd rather wait a few more days and use a 13.6 TeV run for the test. What do you think?

I am not sure at which point there will be also high intensity data, apart from high energy. In any case it's a linear delay, if there's more to fix in the release code everything will get shifted in time.

**Comment by mmusich** [comment]
I am missing feedback on this ticket @tvami . So the preference is just to wait and accumulate delay?

**Comment by tvami** [comment]
I'll make a commit to use 2018 runs (if there is any streamer kept still)

File: etc/ReplayOfflineConfiguration.py [hunk 1]
```diff
@@ -107,9 +108,10 @@
 
 # Configure ScramArch
 setDefaultScramArch(tier0Config, "el8_amd64_gcc10")
+setScramArch(tier0Config, defaultCMSSWVersion['default'], "slc7_amd64_gcc10")
```
Comment (francescobrivio) [inline etc/ReplayOfflineConfiguration.py:111]: @tvami are you sure the changes here are correct for Run 2 data?
It looks different from what we used in the past: https://github.com/dmwm/T0/pull/4674

Comment (tvami) [reply etc/ReplayOfflineConfiguration.py:111]: I'll fix it thanks!

**Comment by germanfgv** [comment]
Testing this right now with 2018pp. You can follow it here:
https://monit-grafana.cern.ch/d/t_jr45h7k/cms-tier0-replayid-monitoring?orgId=11&var-Bin=5m&var-ReplayID=220704122238&var-JobType=All&var-WorkflowType=All

**Comment by tvami** [comment]
hi @germanfgv I see a lot of errors, (1) not sure why prompt ran for many streamers that I thought I had ignored, but ok they are not very relevant for the purpose of this test. (2) on the other hand I see
`Express_Run324841_StreamExpress_Tier0_REPLAY_2022_ID220704123550_v466_220704_1236`
is paused. That's not great, can you please post the error msg?

**Comment by tvami** [comment]
I see this
```
 cms\\nImportError: No module named FWCore.ParameterSet.Config\\n\'\n2022-07-04 14:35:18,360:INFO:Scram:Subprocess stderr was:\nNone\n2022-07-04 14:35:18,360:ERROR:SetupCMSSWPset:Error running scram process. Error code: 5\n2022-07-04 14:35:18,360:ERROR:SetupCMSSWPset:Error creating process for Config/DataProcessing:\nTraceback (most recent call last):\n  File "/srv/job/WMCore.zip/WMCore/WMRuntime/Scripts/SetupCMSSWPset.py", line 695, in __call__\n    self.createProcess(scenario, funcName, funcArgs)\n  File "/srv/job/WMCore.zip/WMCore/WMRuntime/Scripts/SetupCMSSWPset.py", line 156, in createProcess\n    self.scramRun(cmd)\n  File "/srv/job/WMCore.zip/WMCore/WMRuntime/Scripts/SetupCMSSWPset.py", line 119, in scramRun\n    raise RuntimeError(msg)\nRuntimeError: Error running scram process. Error code: 5\n'
```

doesnt make much sense to me. How can there be no `FWCore.ParameterSet.Config` ?

**Comment by tvami** [comment]
I see, for sure the 12_4_X has a different py than then 2018 data. So how do I resolve this?

**Comment by mmusich** [comment]
> I see, for sure the 12_4_X has a different python than the 2018 data. So how do I resolve this?

far from being an expert, but Francesco managed to make it run here: https://github.com/dmwm/T0/pull/4674

**Comment by tvami** [comment]
I had a look at https://github.com/dmwm/T0/pull/4674, there is nothing specific for py2/py3. One difference is that I use el8 for the express while slc7 for the repack. I dont know, maybe something changed behind our back from the T0 side. @germanfgv @jhonatanamado please comment

**Comment by germanfgv** [comment]
I'm not sure what could be causing this. Maybe we can try to run it with slc7 and see if it is different somehow.

**Comment by tvami** [comment]
Ok, let's do that, I'll make a commit soon

**Comment by tvami** [comment]
I think this finished successfully, @mmusich et all, please have a look, https://tinyurl.com/23jlybat

**Comment by mmusich** [comment]
Hi @tvami, yes, superficially looking at the pede exit code things finished well. Honestly I cannot comment from the other plots if the fitted movements make sense.
Eg. from here it's not clear to me why the above significance status bit is not on:
Screenshot from 2022-07-06 10-28-40

even though the payload was uploaded: https://cms-conddb.cern.ch/cmsDbBrowser/logs/condition_uploader_logs/Prep/2022-06-29/2022-07-06/None/None/SiPixelAliHG/None/None/None/any

**Comment by mmusich** [comment]
> OK, is there anything else AlCa can do for you? Otherwise I'll close this PR

I think this PR proves that the new workflow can be safely integrated in the next round of updates of the alca matrix. 
Probably there'll be some adjustements needed after accumulating operational experience with it.

**Comment by tvami** [comment]
Thanks Marco! We'll have this in for the mid-July runs when we move to 12_4_X. Closing the PR now

**Comment by mmusich** [comment]
>  Eg. from here it's not clear to me why the above significance status bit is not on:

for the record, the issue (thanks to @dmeuser) is being tackled here: https://github.com/cms-sw/cmssw/pull/38718 (and it's backport https://github.com/cms-sw/cmssw/pull/38719)

**Pull request closed without merging**